ocaml/xapi/xapi_host_helpers.ml

```diff
@@ -394,17 +407,16 @@ let consider_enabling_host_nolock ~__context =
       else
         f ()
     in
-    if !Xapi_globs.on_system_boot then (
+    if !Xapi_globs.on_system_boot && not (is_startup_complete ()) then (
```

Comment:
`startup_complete` is set at the end of this function, outside of the if-statement that we are inside of here. This means that it will always be set the first time `consider_enabling_host` is called. Now this code is part of an if-statement that checks for other things, like `all_pbds_ok`, which may be `false` the first time `consider_enabling_host` is called. Then you may miss out on clearing the guidance fields.

Now all this interaction with `consider_enabling_host` is getting hard to follow. Also, the function `consider_enabling_host` is meant to enable hosts, and that is not related to clearing guidance DB fields. So overall, I think it is better and simpler to not clear those fields here, but move this to Dbsync_slave, which is meant to sync DB fields when xapi starts, and is guaranteed to be called only when xapi starts.

Reply:
Ah, @robhoes , I see the problem, thank you!

Reply:
@robhoes , thanks for the guidance, I checked the code, now `Dbsync_slave.update_env` is called only when xapi startup. But I think the name `update_env` might be a little confusing, will it be called somewhere else later?
Is it ok to clear the host guidance in `dbsync.setup`? Just after `update_env`. I think maybe the name `setup` is more secure, it should not be called anywhere else except during startup. I had a try: https://github.com/xapi-project/xen-api/pull/5318/commits/79a049e58477269d6ac3f94e5877677fe1745411, please help to review again, thank you!

Reply:
`Dbsync_slave.update_env` is the right place to do this sort of thing.

Reply:
OK, changed: https://github.com/xapi-project/xen-api/pull/5318/commits/a3028e1d8677207c8e52f07625887a3f815bc685